libraries/rush-lib/src/logic/operations/ShellOperationRunner.ts

```diff
@@ -420,7 +450,7 @@ export class ShellOperationRunner implements IOperationRunner {
 
         const [, cacheWriteSuccess] = await Promise.all([writeProjectStatePromise, setCacheEntryPromise]);
 
-        if (buildCacheTerminalProvider.hasErrors) {
+        if (terminalProvider.hasErrors) {
```

Comment:
I'm pretty sure this was right. This check is in the cache write code path; if the build failed or wrote to stderr that was already handled outside of the parent `if` check on line 425.